app/hotels/src/filter/score/ScorePopup.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import { ButtonPopup, Slider } from '@kiwicom/react-native-app-common';
+
+type Props = {|
+  onClose: () => void,
+  onSave: (minScore: number | null) => void,
+  isVisible: boolean,
+  minScore: number | null,
+|};
+
+type State = {|
+  minScore: number,
+|};
+
+const SLIDER_SHIFT = 5;
+
+export default class ScorePopup extends React.Component<Props, State> {
+  state = {
+    minScore: 0,
+  };
+
+  componentWillReceiveProps = ({ minScore }: Props) =>
+    this.setState({ minScore: minScore ? minScore - SLIDER_SHIFT : 0 }); // Normalize null/6-9 options to slider's range 0-4
```

Comment:
Instead of the comment, you can introduce a well named function. Something like convertScoreToSliderRange.